### ISSUE TITLE
Fixing Documentation for Node-Level AttentionalAggregation 

### ISSUE THREAD

**Issue by RafiBrent** [pyg-team/pytorch_geometric#5471]
### 📚 Describe the documentation issue

Sorry to reopen issue #5400, but I wanted to check the exact function of the node-level gating implementation. Based on the equation in the docs, it looks like the output of h_gate on each set of node features is softmaxed individually, but if h_gate reduces each set of node features to a scalar then this would simply return 1, reverting the whole process to an unweighted sum of the outputs of h_theta. On the other hand, true node-level gating would require softmaxing the entire collection of outputs of h_gate (a vector of length N_i) and then using the elements of this vector to weight the corresponding output vectors of h_theta before summing them all together. Please let me know if I'm missing something here, but I believe that the documentation or the code need to be modified for node-level attentional aggregation.

**Comment by Padarn** [comment]
Hey @RafiBrent could you explain in a bit more detail what you would expect to see in the formula? The softmax is currently written as 

> \mathrm{softmax} \left(h_{\mathrm{gate}} ( \mathbf{x}_n ) \right) 

As I understand it, the softmax here is over features from the incoming nodes. This seems to match the approach in the paper.

**Comment by RafiBrent** [comment]
Hi Padarn,
   Thanks so much for your reply. Yes, the formula given does currently match the approach used in the paper, but this is for the newly-implemented feature-level aggregation which Hadamard multiplies the output of h_gate (length out_channels) by the output of h_theta. However, the original implementation (what is now referred to as "node-level gating" in the docs) stipulated that h_gate output a vector of length one, which is different from what is done in the literature. My question is how the softmax is done in this case-- is it really softmaxing the scalar output of h_gate applied to x_n (which would return 1 regardless of this output's value), or is it softmaxing the collection of all such scalar outputs (length N_i)?

**Comment by Padarn** [comment]
Hi @RafiBrent. I see, thanks for giving more details.
>  is it really softmaxing the scalar output of h_gate applied to x_n
No. Here is the implementation
```
        gate = self.gate_nn(x)
        x = self.nn(x) if self.nn is not None else x
        gate = softmax(gate, index, ptr, dim_size, dim)
        return self.reduce(gate * x, index, ptr, dim_size, dim)
```
So, as you say, the softmax is taken over the output of `gate(x)`, not `gate(x_n)`. 

Looking again at the paper, they do not actually take a softmax, but rather just apply a sigmoid.

**Comment by RafiBrent** [comment]
Thanks so much for the clarification. In terms of the docs, I think that it would probably be clearer to include two different equations, one for feature-level aggregation (same as currently written) and one for node-level aggregation. This second equation wouldn't need a summation and could be written as matrix multiplication of h_theta(x) and the softmaxed output of h_gate(x). Thanks so much for your help with this!
